### PR TITLE
Add makefile targets for manila kuttl tests

### PR DESCRIPTION
**Pull request by vhariria**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1094,6 +1096,21 @@ glance_kuttl: namespace input openstack_crds deploy_cleanup mariadb mariadb_depl
 	make deploy_cleanup
 	make cleanup
 
+.PHONY: manila_kuttl_run
+manila_kuttl_run:
+   INSTALL_YAMLS=${INSTALL_YAMLS} kubectl-kuttl test --config ${MANILA_KUTTL_CONF} ${MANILA_KUTTL_DIR}
```
Comment (frenzyfriday) [inline Makefile:1101]: This has to start with Tab character

**Review by frenzyfriday — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -1094,6 +1096,21 @@ glance_kuttl: namespace input openstack_crds deploy_cleanup mariadb mariadb_depl
...
+
+.PHONY: manila_kuttl
+manila_kuttl: namespace input openstack_crds deploy_cleanup mariadb mariadb_deploy rabbitmq rabbitmq_deploy keystone_deploy_prep keystone keystone_deploy manila_deploy_prep manila infra mariadb_deploy_validate ## runs kuttl tests for the manila operator. Installs openstack crds and manila operators and cleans up previous deployments before running the tests and, add cleanup after running the tests.
```
[the remark this answers is not in the document]
Comment (vhariria) [reply Makefile]: Done, thanks.

File: Makefile [hunk 3]
```diff
@@ -1094,6 +1096,21 @@ glance_kuttl: namespace input openstack_crds deploy_cleanup mariadb mariadb_depl
...
+   make manila_kuttl_run
+   make infra_cleanup
+   make rabbitmq_deploy_cleanup
+   make rabbitmq_cleanup
+   make deploy_cleanup
+   make manila_cleanup
+   make keystone_cleanup
+   make mariadb_cleanup
```
[the remark this answers is not in the document]
Comment (vhariria) [reply Makefile]: Verified in local env, thanks.

File: Makefile (hunk 1 above)
Comment (vhariria) [reply Makefile]: Verified in local env, thanks.

File: Makefile [hunk 4]
```diff
@@ -1345,6 +1347,18 @@ glance_kuttl: input openstack_crds deploy_cleanup mariadb mariadb_deploy keyston
...
+manila_kuttl_run: ## runs kuttl tests for the manila operator,
+	kubectl-kuttl test --config ${MANILA_KUTTL_CONF} ${MANILA_KUTTL_DIR}
+
+.PHONY: manila_kuttl
+manila_kuttl: namespace input openstack_crds deploy_cleanup mariadb mariadb_deploy keystone keystone_deploy infra ceph manila manila_deploy ## runs kuttl tests for manila operator. Installs openstack crds and manila operators and cleans up previous deployments before running the tests and, add cleanup after running the tests.
+	$(eval $(call vars,$@,manila))
+	make wait
+	make manila_kuttl_run
+	make deploy_cleanup
+	make cleanup
```
Comment (cescgina) [inline Makefile:1360]: we should add `ceph_cleanup` here as well

File: Makefile [hunk 5]
```diff
@@ -1345,6 +1347,18 @@ glance_kuttl: input openstack_crds deploy_cleanup mariadb mariadb_deploy keyston
 	make deploy_cleanup
 	make cleanup
 
+.PHONY: manila_kuttl_run
+manila_kuttl_run: ## runs kuttl tests for the manila operator,
+	kubectl-kuttl test --config ${MANILA_KUTTL_CONF} ${MANILA_KUTTL_DIR}
+
+.PHONY: manila_kuttl
+manila_kuttl: namespace input openstack_crds deploy_cleanup mariadb mariadb_deploy keystone keystone_deploy infra ceph manila manila_deploy ## runs kuttl tests for manila operator. Installs openstack crds and manila operators and cleans up previous deployments before running the tests and, add cleanup after running the tests.
```
Comment (cescgina) [inline Makefile:1355]: It would probably be a good idea to introduce the changes to run the tests in a namespace different than `openstack` here

Comment (vhariria) [reply Makefile:1355]: Thanks Joan, I am planning to make the changes once this PR merges.

File: devsetup/Makefile [hunk 6]
```diff
@@ -8,7 +8,7 @@ EDPM_TOTAL_NODES ?= 1
 BMAAS_INSTANCE_NAME_PREFIX ?= crc-bmaas
 BMAAS_INSTANCE_MEMORY ?= 4096
 BMAAS_INSTANCE_VCPUS ?= 2
-BMAAS_INSTANCE_DISK_SIZE ?= 10
+BMAAS_INSTANCE_DISK_SIZE ?= 20
```
Comment (viroel) [inline devsetup/Makefile:11]: why do you need this change for kuttl tests? this seems to be used only for BMAAS scenarios.

File: devsetup/scripts/bmaas/vbm-setup.sh [hunk 7]
```diff
@@ -20,7 +20,7 @@ ACTION=""
 # Virtual Machine spec
 MEMORY=${MEMORY:-4096}
 VCPUS=${VCPUS:-2}
-DISK_SIZE=${DISK_SIZE:-10}
+DISK_SIZE=${DISK_SIZE:-20}
```
Comment (viroel) [inline devsetup/scripts/bmaas/vbm-setup.sh:23]: ditto.

File: devsetup/Makefile (hunk 6 above)
Comment (vhariria) [reply devsetup/Makefile]: Thanks, resolved.

File: devsetup/scripts/bmaas/vbm-setup.sh (hunk 7 above)
Comment (vhariria) [reply devsetup/scripts/bmaas/vbm-setup.sh]: Ditto, thanks.

**Comment by cescgina** [comment]
recheck

**Comment by cescgina** [comment]
recheck

**Comment by vhariria** [comment]
recheck

**Comment by vhariria** [comment]
recheck

File: Makefile [hunk 8]
```diff
@@ -511,15 +514,15 @@ openstack_cleanup: operator_namespace## deletes the operator, but does not clean
 
 .PHONY: openstack_deploy_prep
 openstack_deploy_prep: export KIND=OpenStackControlPlane
-openstack_deploy_prep: openstack_deploy_cleanup netconfig_deploy ## prepares the CR to install the service based on the service sample file OPENSTACK
+openstack_deploy_prep: openstack_deploy_cleanup ## prepares the CR to install the service based on the service sample file OPENSTACK
```
Comment (gouthampacha) [inline Makefile:517]: think this is a bad rebase?

File: Makefile [hunk 9]
```diff
@@ -685,7 +688,8 @@ netconfig_deploy: input netconfig_deploy_prep ## installs the service instance u
 .PHONY: netconfig_deploy_cleanup
 netconfig_deploy_cleanup: ## cleans up the service instance, Does not affect the operator.
 	$(eval $(call vars,$@,infra))
-	oc delete --all netconfig --ignore-not-found=true || true
+	oc kustomize ${DEPLOY_DIR} | oc delete --ignore-not-found=true -f -
+	rm -Rf ${OPERATOR_BASE_DIR}/infra-operator ${DEPLOY_DIR}
```
Comment (gouthampacha) [inline Makefile:692]: same comment on this line and the change to `netconfig_deploy` above..

File: Makefile [hunk 10]
```diff
@@ -1473,6 +1477,21 @@ glance_kuttl: kuttl_common_prep glance glance_deploy_prep ## runs kuttl tests fo
...
+manila_kuttl_run: ## runs kuttl tests for the manila operator,
+	kubectl-kuttl test --config ${MANILA_KUTTL_CONF} ${MANILA_KUTTL_DIR}
+
+.PHONY: manila_kuttl
+manila_kuttl: kuttl_common_prep ceph manila manila_deploy_prep ## runs kuttl tests for manila operator. Installs manila operator and cleans up previous deployments before and after running the tests.
+	$(eval $(call vars,$@,manila))
+	make wait
+	make manila_kuttl_run
+	make deploy_cleanup
+	make manila_cleanup
```
Comment (gouthampacha) [inline Makefile:1490]: nit: cleanup in the reverse order of install; `manila_cleanup` before `deploy_cleanup`

File: scripts/gen-ceph-kustomize.sh [hunk 11]
```diff
@@ -41,6 +41,7 @@ POOLS=("volumes" "images" "backups" "cephfs.cephfs.meta" "cephfs.cephfs.data")
 DAEMONS="osd,mds"
 DATA_SIZE=${DATA_SIZE:-500Mi}
 WORKER=${WORKER:-""}
+MON_CONF=${MON_CONF:-""}
```
Comment (gouthampacha) [inline scripts/gen-ceph-kustomize.sh:44]: looks like a bad rebase, no changes seem necessary on this file

File: scripts/gen-input-kustomize.sh [hunk 12]
```diff
@@ -74,11 +74,8 @@ secretGenerator:
   - OctaviaDatabasePassword=${PASSWORD}
   - NovaPassword=${PASSWORD}
   - NovaAPIDatabasePassword=${PASSWORD}
```
Comment (gouthampacha) [inline scripts/gen-input-kustomize.sh:76]: these changes seem unrelated as well, please rebase against main again..

**Review by gouthampacha — CHANGES_REQUESTED** [review]
Hi Vida; this change has some rebase issues.. comments inline

**Review by gouthampacha — APPROVED** [review]
This looks good to me; thanks!

**Comment by viroel** [comment]
recheck
